src/gtk/control.cpp

```diff
@@ -341,9 +341,20 @@ wxSize wxControl::GTKGetPreferredSize(GtkWidget* widget) const
 #ifdef __WXGTK3__
     int w, h;
     gtk_widget_get_size_request(widget, &w, &h);
+    // gtk_widget_get_preferred_size is not works correctly for the hidden contol.
+    // So workaround this case.
+    const bool hiddenAtStart = !gtk_widget_get_visible(widget);
+    if ( hiddenAtStart )
+    {
+        gtk_widget_show(widget);
```

Comment:
I'm pretty uncomfortable with this. It's hard to say what sort of unintended effects might result from forcing a hidden widget to be temporarily shown. At a minimum, transitory visual glitches  would not surprise me.